Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchSubmission.java

```
@@ -74,8 +88,25 @@ public String getInstallationID() {
 
     @NotNull
     public CloseableHttpClient getHttpClient() throws IOException {
+        TrustStrategy ts = ServiceManager.getServiceProvider(TrustStrategy.class);
+        SSLConnectionSocketFactory sslSocketFactory = null;
+
+        if (ts != null) {
+            try {
+                SSLContext sslContext = new SSLContextBuilder()
+                        .loadTrustMaterial(ts)
+                        .build();
+
+                //TODO:add by pass option,if bypass use noophostnameverifier else defaulthostnameverifier
+                sslSocketFactory = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
+            } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException e) {
+                log().warn("Prepare SSL Context for HTTPS failure", e);
```

I suppose the exception should not be swallowed.